### ISSUE TITLE
Inconsistent use of id's when deleting metadata

### ISSUE THREAD

**Issue by mortenwh** [metno/discovery-metadata-catalog-ingestor#158]
I tried first deleting with `no.met:<uuid>`. This worked. Then I tried with only `<uuid>` as argument. Then the pycsw distributor failed. Repeating this with `no.met:` included, resulted in a failing file distributor deletion. So, it seems the file distributor works regardless of whether the namespace is included or not, whereas the pycsw requires the namespace. This needs to be consistent.

To do:

- [ ] Decide how we want to call the delete function
- [ ] Make tests that ensure consistency between the distributors on all API calls (create, update and delete)
- [ ] Make the delete call, and possibly other calls consistent

Example:

```
(production-12-2021) mortenwh@ppi-r8login-b1:~$ curl -X POST https://dmci.s-enda-dev.k8s.met.no/v1/delete/no.met:d3bdbbb4-c1bf-45ae-aedb-88fbe02a721f
Everything is OK
(production-12-2021) mortenwh@ppi-r8login-b1:~$ curl -X POST https://dmci.s-enda-dev.k8s.met.no/v1/delete/fe5af8d5-2767-4e44-994c-e4f59c688f81
The following distributors failed: pycsw
 - pycsw: <?xml version="1.0" encoding="UTF-8" standalone="no"?>
<!-- pycsw 2.7.dev0 -->
<csw:TransactionResponse xmlns:csw="http://www.opengis.net/cat/csw/2.0.2" xmlns:dc="http://purl.org/dc/elements/1.1/" xmlns:dct="http://purl.org/dc/terms/" xmlns:gmd="http://www.isotc211.org/2005/gmd" xmlns:gml="http://www.opengis.net/gml" xmlns:ows="http://www.opengis.net/ows" xmlns:xs="http://www.w3.org/2001/XMLSchema" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" version="2.0.2" xsi:schemaLocation="http://www.opengis.net/cat/csw/2.0.2 http://schemas.opengis.net/csw/2.0.2/CSW-publication.xsd"><csw:TransactionSummary><csw:totalInserted>0</csw:totalInserted><csw:totalUpdated>0</csw:totalUpdated><csw:totalDeleted>0</csw:totalDeleted></csw:TransactionSummary></csw:TransactionResponse>
(production-12-2021) mortenwh@ppi-r8login-b1:~$ curl -X POST https://dmci.s-enda-dev.k8s.met.no/v1/delete/no.met:fe5af8d5-2767-4e44-994c-e4f59c688f81
The following distributors failed: file
 - file: File not found: fe5af8d5-2767-4e44-994c-e4f59c688f81.xml
```

**Comment by magnarem** [comment]
I would say that the `id` _always should be_ `<ns>:<uuid>` ie. the same as the `<mmd:metadata_identifier>`-element in the MMD. 

This will also make the easy for a data producer. So if a data producer sees in the data-portal search interface `data.met.no` that a dataset is in the catalog that should not be there, the search-interface will present the `metadata_identifier` in solr index, that will be the same as `<ns:>:<uuid>` since we defined that a metadata_identifier always are on this form.

**Comment by magnarem** [comment]
See my comment in #171.

[issue #171 comment](https://github.com/metno/discovery-metadata-catalog-ingestor/issues/171#issuecomment-1564050923)

**Comment by mortenwh** [comment]
Solved in #171
